views/challenge.py

```diff
@@ -108,7 +137,7 @@ def _get_challenges_groupby_level(self) -> dict[Level, list[ChallengeName]]:
             challenge_names.sort()
 
         # Make sure groups are ordered by level (from easy to hard)
-        return {level: groups[level] for level in Level}
+        return {level: groups.get(level, []) for level in Level}
```

Comment:
I would suggest changing groups to a defaultdict, so that we can remove the `groups.setdefault` above and the `.get(level, [])` here

Reply:
Make sense, updated.